src/codeu/chat/client/simplegui/ChatGuiFX.java

```diff
@@ -283,107 +302,137 @@ private void signUpButtonClicked(ActionEvent e) {
+    * to the list of conversations to be displayed.
+    */
     private void addConversation(ActionEvent e) {
         if (clientContext.user.hasCurrent()) {
             // popup for the user to add a conversation
             TextInputDialog dialog = new TextInputDialog("Name your conversation!");
-            dialog.setTitle(" "); // make the title blank
+            dialog.setTitle(" ");
             dialog.setHeaderText("Create your conversation");
 
             // get input and add the name of the convo
             String name = dialog.showAndWait().get();
 
             // TODO: check for duplicate conversations & handle them
 
-            if (name != null && name.length() > 0) {
+            if (!name.isEmpty() && name.length() > 0) {
                 clientContext.conversation.startConversation(name, clientContext.user.getCurrent().id);
                 convoList.addAll(name);
             }
         } else {
             // user is not signed in
-            Alert alert = new Alert(AlertType.ERROR);
-            alert.setHeaderText("You're not signed in!");
-            alert.showAndWait();
+            displayAlert("You're not signed in!");
         }
     }
 
+    /**
+    * When the user clicks on a conversation in the right panel, it gets its index and name to
+    * fetch its contents, which is a ConversationSummary object. Then it sets the current conversation
+    * to the selected conversation.
+    */
     private void selectConversation(MouseEvent e) {
         // set the conversation title
         int index = conversations.getSelectionModel().getSelectedIndex();
         String data = conversations.getSelectionModel().getSelectedItem();
-        ConversationSummary cs = ChatGuiFX.this.lookupByTitle(data, index);
-
-        clientContext.conversation.setCurrent(cs);
-        update(cs);
+        // get contents of conversation
+        ConversationSummary selectedConvo = lookupByTitle(data, index);
+        // set new conversation
+        clientContext.conversation.setCurrent(selectedConvo);
+        updateCurrentConversation(selectedConvo);
     }
 
-    // Locate the Conversation object for a selected title string.
-    // index handles possible duplicate titles.
+    /**
+    * Finds the ConversationSummary (the contents of the conversation) that corresponds to
+    * the selected conversation
+    * @param title  name of the selected conversation
+    * @param index  index of selected conversation in the list of conversations
+    */
     private ConversationSummary lookupByTitle(String title, int index) {
-
-      int localIndex = 0;
-      for (final ConversationSummary cs : clientContext.conversation.getConversationSummaries()) {
-        if ((localIndex >= index) && cs.title.equals(title)) {
-          return cs;
+        int localIndex = 0;
+        for (final ConversationSummary cs : clientContext.conversation.getConversationSummaries()) {
+            if ((localIndex >= index) && cs.title.equals(title)) {
```

Comment:
Yeah, I'd think about using a `Map` for this. It's amazing what you can solve with a `HashMap`!